lib/_debugger.js

```diff
@@ -1639,7 +1639,16 @@ Interface.prototype.trySpawn = function(cb) {
   } else if (this.args.length === 3) {
     // `node debug -p pid`
     if (this.args[1] === '-p' && /^\d+$/.test(this.args[2])) {
-      process._debugProcess(parseInt(this.args[2], 10));
+      const pid = parseInt(this.args[2], 10);
+      try {
+        process._debugProcess(pid);
+      } catch (e) {
+        if (e.code === 'ESRCH') {
+          console.error('Target process: ' + pid + ' doesn\'t exist.');
```

Comment:
Didn’t see template string be used in node core, so I don’t use it. If it’s recommend, I will replace it.

> 在 2015年6月5日，上午11:21，Yosuke Furukawa <notifications@github.com> 写道：
> 
> In lib/_debugger.js <https://github.com/nodejs/io.js/pull/1863#discussion_r31785321>:
> 
> > @@ -1639,7 +1639,16 @@ Interface.prototype.trySpawn = function(cb) {
> >    } else if (this.args.length === 3) {
> >      // `node debug -p pid`
> >      if (this.args[1] === '-p' && /^\d+$/.test(this.args[2])) {
> > -      process._debugProcess(parseInt(this.args[2], 10));
> > +      const pid = parseInt(this.args[2], 10);
> > +      try {
> > +        process._debugProcess(pid);
> > +      } catch (e) {
> > +        if (e.code === 'ESRCH') {
> > +          console.error('Target process: ' + pid + ' doesn\'t exist.');
> We don't need to use escape \ enclosed by " or `.
> 
> console.error(`Target process: ${pid} doesn't exist.`);
> —
> Reply to this email directly or view it on GitHub <https://github.com/nodejs/io.js/pull/1863/files#r31785321>.
>